### PR TITLE
Finish up the last 3 non-implications for the original 11 equalities

### PR DESCRIPTION
**Pull request by amirlb**

### CODE REVIEW

File: equational_theories/Basic.lean [hunk 1]
```diff
@@ -188,6 +188,18 @@ theorem Equation43_not_implies_Equation42 : ∃ (G: Type) (_: Magma G), Equation
...
 
+theorem Equation43_not_implies_Equation387 : ∃ (G: Type) (_: Magma G), Equation43 G ∧ ¬ Equation387 G := by
+  let hG : Magma Nat := {
+    op := fun x y => x+y
+  }
+  use ℕ, hG
+  constructor
+  . exact Nat.add_comm
+  by_contra h
+  replace h := h 0 1
```
Comment (Shreyas4991) [inline equational_theories/Basic.lean:199]: You can use `specialize h 0 1` here

File: equational_theories/Basic.lean [hunk 2]
```diff
@@ -245,7 +257,26 @@ theorem Equation4512_not_implies_Equation42 : ∃ (G: Type) (_: Magma G), Equati
...
+  let hG : Magma Real := {
+    op := fun x y => (x + y) / 2
+  }
+  use ℝ, hG
+  constructor
+  . intro x y
+    simp [hG]
+    ring
+  by_contra h
+  replace h := h 0 0 1
```
Comment (Shreyas4991) [inline equational_theories/Basic.lean:281]: `specialize h 0 0 1` also works in place of `replace h := h 0 0 1` and is shorter

**Review by Shreyas4991 — COMMENTED** [review]
The comments are merely aesthetic suggestions to shorten a couple of lines. Overall it seems fine.

**Review by Shreyas4991 — APPROVED** [review]
I'll leave the aesthetic tactic choices to @teorth This PR builds. Somehow the build task wasn't triggered on this PR. This probably needs checking.

File: equational_theories/Basic.lean (hunk 1 above)
Comment (amirlb) [reply equational_theories/Basic.lean:199]: Looks like elsewhere in the file we used the form `replace h := h ...` too. Shall I replace all of them?

Comment (teorth) [reply equational_theories/Basic.lean:199]: Sure, we can do that in a separate PR

Comment (teorth) [reply equational_theories/Basic.lean:199]: I've gone ahead and implemented the change globally.

**Pull request merged**